core/services/ping/pingmanager.py

```diff
@@ -1,38 +1,67 @@
+import asyncio
 import logging
-from typing import Dict, List
+from typing import Any, Dict, List
 
 from serial import Serial
 
 from ping1d_driver import Ping1DDriver
 from ping360_driver import Ping360Driver
 from pingdriver import PingDriver
-from pingutils import PingDeviceDescriptor, PingType
+from pingutils import PingDeviceDescriptor, PingType, udp_port_is_in_use
 
 
 class PingManager:
     def __init__(self) -> None:
         self.drivers: Dict[PingDeviceDescriptor, PingDriver] = {}
-        self.ping1d_current_port: int = 9090
-        self.ping360_current_port: int = 9092
+        self.ping1d_base_port: int = 9090
+        self.ping360_base_port: int = 9092
 
     def stop_driver_at_port(self, port: Serial) -> None:
         """Stops the driver instance running for port "port" """
         ping_at_port = list(filter(lambda ping: ping.port == port, self.drivers.keys()))
```

Comment:
Clean up with list comprehension. Default dictionary iterator is already over the keys.
```suggestion
        ping_at_port = [ping for ping in self.drivers if ping.port == port]
```